content/chat/participant-activity.textile

```diff
@@ -1,7 +1,223 @@
+
+```[realtime_javascript]
+channel.presence.leave((err) => {
+  // Custom functionality to update UI for chat participant that they are no longer online
+});
+```
+
+When a chat participant triggers the @leave@ event all subscribers for this presence channel will be notified in real-time allowing them to update the visual online status of this user for everyone.
+
+```[realtime_javascript]
+channel.presence.subscribe(‘leave’, (member) => {
+  user.online = false
+  user.lastOnline = member.timestamp
+
+  // Custom functionality to update display for chat participant that another participant is no longer online
+});
+```
+
+h3(#update). Update
+
+In certain chat instances, chat members can associate their chat participant status with a text-based status. For example, if you have a chat participant object (user) with the @statusMessage@ field, you can trigger an @update@ event using the code sample below to update this user with a new text-based string.
+
+```[realtime_javascript]
+channel.presence.update('<NEW_STATUS_UPDATE>', (err) => {
+  user.statusMessage = '<NEW_STATUS_UPDATE>';
+});
+```
+
+When a chat participant triggers the @update@ event all subscribers for this presence channel will be notified in real-time allowing them to update the string value of that chat participant's status of this user for everyone.
+
+```[realtime_javascript]
+channel.presence.subscribe('update', (memberUpdate) => {
+  user.status = memberUpdate.data
+});
+```
+
+h3(#other-events). Other events / more information
+
+The @present@ event is not relevant for chat participant statuses, however, if you wish to read more about other functionalities of presence, please refer to the "presence":/presence-occupancy/presence?lang=javascript#trigger-events docs page.
+
+h2(#subscribe). Subscribe chat participants to events
+
+Chat participant clients can subscribe to the presence set, enabling them to receive updates on other participants and update the UI accordingly (e.g., displaying online status icons).
+
+The message count within a presence set, like online status, can surge as the participant number grows. For instance, with 200 participants rapidly joining and leaving, the potential messages for online status alone could reach 80,400. The breakdown includes:
+
+200 chat participant messages published for the @enter@ event.
+200 × 200 (40,000) messages subscribed to for the @enter@ events.
+200 chat participant messages published for the @leave@ event.
+200 × 200 (40,000) presence messages subscribed to for the @leave@ event.
+
+For improved scalability, consider making your backend server the exclusive subscriber to chat participants' online status updates if your chat application anticipates a large number of participants. In this scenario, frontend clients can periodically request updates from the backend.
+
```

Comment:
We could also recommend adding only a subset of all users to the member data and using occupancy metrics to indicate the total number of listeners (aka Spotify implementation).

Reply:
https://ably.atlassian.net/browse/CON-109

As discussed, I've created a ticket to look into this when we revisit these docs for large scale focus.